web/src/composebox_typeahead.ts

```diff
@@ -954,29 +955,39 @@ export function get_candidates(
                     topic_list.push(token);
                 }
                 const matcher = get_topic_matcher(token);
+                // We want the empty topic's display name to match the query
+                // while searching for topics in the typeahead.
+                const empty_topic_idx = topic_list.indexOf("");
+                if (empty_topic_idx !== -1) {
+                    topic_list[empty_topic_idx] = util.get_final_topic_display_name("");
+                }
                 const matches = topic_list.filter((item) => matcher(item));
-                const matches_list: TopicSuggestion[] = matches.map((topic) => ({
-                    topic,
-                    type: "topic_list",
-                    is_channel_link: false,
-                    stream_data: {
-                        ...sub,
-                        type: "stream",
-                        // The channel description is only rendered for the channel mention
-                        // itself, not topic rows, so we leave this blank.
-                        rendered_description: "",
-                    },
-                }));
+                const matches_list: TopicSuggestion[] = matches.map(
+                    (topic, current_topic_index) => ({
+                        topic_display_name: util.get_final_topic_display_name(topic),
```

Comment:
Here `topic` is already a `topic_display_name`.

What role `util.get_final_topic_display_name` plays here ? -- let me know if I'm missing something.